lib/galaxy/jobs/runners/local.py

```diff
@@ -24,6 +24,20 @@
 
 log = logging.getLogger(__name__)
 
+gpu_flag = 0
+bash_command = "/bin/bash -c 'nvidia-smi'"
+sp = subprocess.Popen(bash_command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+out, err = sp.communicate()
+command_not_found = "command not found"
+if command_not_found.encode() not in out and command_not_found.encode() not in err:
+    # log.debug("LOCAL: out: %s, err: %s, command_not_found.encode: %s COMMAND NOT FOUND NOT IN OUT" % (out, err, command_not_found.encode()))
+    gpu_flag = 1
+    import pynvml as nvml
+
+if gpu_flag == 1:
+    nvml.nvmlInit()
+    gpu_count = nvml.nvmlDeviceGetCount()
```

Comment:
```suggestion
def get_local_gpu_count():
  bash_command = "/bin/bash -c 'nvidia-smi'"
  sp = subprocess.Popen(bash_command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
  out, err = sp.communicate()
  command_not_found = "command not found"
  if command_not_found.encode() not in out and command_not_found.encode() not in err:
      # log.debug("LOCAL: out: %s, err: %s, command_not_found.encode: %s COMMAND NOT FOUND NOT IN OUT" % (out, err, command_not_found.encode()))
      import pynvml as nvml
      nvml.nvmlInit()
      return nvml.nvmlDeviceGetCount()
```

Reply:
Hi, I addressed these comments in my latest commit. I also got rid of the usage of pynvml and did it with regular Python (so it won't need additional libraries). 
After this commit, I will work on adding the multi-gpu scheduling according to GPU memory usage.